doc/kinsol/guide/source/Usage/index.rst

```diff
@@ -2132,6 +2147,59 @@ from :numref:`KINSOL.Mathematics.FixedPoint` and
      a non-zero value if an error occurred.
 
 
+.. _KINSOL.Usage.CC.user_fct_sim.depthFn:
+
+Depth function
+~~~~~~~~~~~~~~
+
+When using the fixed-point or Picard iterations, the user may provide a function
+of type :c:type:`KINDepthFn` to modify the Anderson acceleration depth,
+:math:`m_n`, from :numref:`KINSOL.Mathematics.AndersonAcceleration`.
+
+.. c:type:: int (*KINDepthFn)(long int iter, N_Vector u_val, N_Vector g_val, N_Vector f_val, N_Vector* df, sunrealtype* R_mat, long int depth, void* user_data, long int* new_depth, long int* remove_indices);
+
+   This function computes an Anderson acceleration depth parameter, :math:`0
+   \leq \hat{m}_n \leq \min\{m_n, m_\textrm{max}\}` to replace the current
+   depth, :math:`m_n`, where :math:`m_\textrm{max}` is the maximum
+   acceleration space size (see :c:func:`KINSetMAA`).
+
+   If the new depth computed by this function, :math:`\hat{m}_n`, is less than
+   the current depth, :math:`m_n`, then the :math:`m_n - \hat{m}_n` left-most
+   columns of :math:`\Delta F_n` will be removed and the QR factorization
+   updated accordingly. Otherwise, the current depth and factorization will be
+   retained.
+
+   **Parameters:**
+
+   * **iter** -- the iteration being computed, :math:`n + 1`.
+   * **u_val** -- the current iterate, :math:`u_n`.
+   * **g_val** -- the fixed-point function evaluated at the current iterate,
+     :math:`G(u_n)`.
+   * **f_val** -- the fixed-point residual function evaluated at the current
+     iterate, :math:`f_n = G(u_n) - u_n`.
+   * **df** -- the history of fixed-point residual function differences,
+     :math:`\Delta F_{n} = (\Delta f_{n-m_n}, \ldots, \Delta f_{n-1})`, where
+     :math:`\Delta f_i = f_{i+1} - f_i`.
+   * **R_mat** -- the upper triangular matrix, :math:`R_n`, in the QR
+     factorization :math:`\Delta F_n = Q_n R_n`. ``R_mat`` is allocated with
+     space for a dense :math:`m_{\textrm{max}} \times m_{\textrm{max}}` matrix
+     and elements are stored column-wise. On input, ``R_mat`` contains the
+     entries for an :math:`m_n \times m_n` matrix.
+   * **depth** -- the current size of the Anderson acceleration space,
+     :math:`m_n`.
+   * **user_data** -- the user data pointer passed to :c:func:`KINSetUserData`.
+   * **new_depth** -- the computed depth, :math:`\hat{m}_n`, to replace
+     the current depth, :math:`m_n`.
+   * **remove_indices** -- this parameter is currently unused (``NULL``) and
+     will be used to provide support for removing specific columns of
+     :math:`\Delta F_n` in the future.
```

Comment:
I'm not familiar with how this is intended to be used in the future, but I'm surprised the type is `long int *` and not `sunbooleantype *` to indicate which to keep and remove.

Reply:
I was thinking it would an array with which indices to remove but using `sunbooleantype *` would be clearer. I'll switch to that.